From 02266242494e2fb5a6f722222a5d69b9451912ac Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Thu, 22 Feb 2024 17:11:57 +0100
Subject: [PATCH] Make tests parallel safe
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

If running tests in parallel, they often fail because t/00setup.t
needs to complete before other tests:

    $ HARNESS_OPTIONS=j4 BACKPAN_INDEX_TEST_NO_INTERNET=1 ./Build test
    ===(       4;0   1/11  1/4  1/?   1/84 )================================# Fetching the BackPAN index and creating the database. This may take a while.
    DBD::SQLite::st execute failed: UNIQUE constraint failed: files.path at /tmp/BackPAN-Index/blib/lib/BackPAN/Index.pm line 217, <$_[...]> line 2.
    DBD::SQLite::st execute failed: UNIQUE constraint failed: files.path at /tmp/BackPAN-Index/blib/lib/BackPAN/Index.pm line 217, <$_[...]> line 2.
    Issuing rollback() due to DESTROY without explicit disconnect() of DBD::SQLite::db handle dbname=/tmp/BackPAN-Index/t/cache/backpan.sqlite at /tmp/BackPAN-Index/blib/lib/BackPAN/Index.pm line 217.
    # Looks like your test exited with 2 just after 1.
    t/00setup.t ........................ Dubious, test returned 2 (wstat 512, 0x200)
    Failed 3/4 subtests

This patch adds TAP::Harness test rules so that t/00setup.t is run always first.

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 t/testrules.yml | 3 +++
 1 file changed, 3 insertions(+)
 create mode 100644 t/testrules.yml

diff --git a/t/testrules.yml b/t/testrules.yml
new file mode 100644
index 0000000..b0351a6
--- /dev/null
+++ b/t/testrules.yml
@@ -0,0 +1,3 @@
+seq:
+  - seq: t/00setup.t
+  - par: **
-- 
2.43.2

